### PR TITLE
refactor(meta): add dedicated error enum for hummock meta.

### PR DESCRIPTION
**Pull request by zwang28**
## What's changed and what's your intention?

Previously hummock meta module use RwError directly. This PR defines its dedicated error enum, which allows to distinguish recoverable error (e.g. meta store failure) from permanent error (context invalidation).

TODO:
- Current mapping from etcd error to meta store error is not enough to distinguish recoverable error from permanent error.

### CODE REVIEW

File: src/meta/src/hummock/error.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+impl From<meta_store::Error> for Error {
+    fn from(error: meta_store::Error) -> Self {
+        match error {
+            meta_store::Error::ItemNotFound(err) => Error::InternalError(err),
+            meta_store::Error::TransactionAbort() => {
+                // TODO: need more concrete error from meta store.
+                Error::InvalidContext(0)
+            }
+            meta_store::Error::Internal(err) => Error::MetaStoreError(err),
```
Comment (hzxa21) [inline src/meta/src/hummock/error.rs:51]: Why do we convert Internal error to `MetaStoreError` (retryable)? Converntionally, internal error is normally not retryable.

File: src/meta/src/hummock/hummock_manager.rs [hunk 2]
```diff
@@ -595,21 +591,16 @@ where
...
 
-        if should_commit {
-            commit_multi_var!(
-                self,
-                Some(assignee_context_id),
-                compact_status,
-                compact_task_assignment
-            )?;
-        } else {
-            abort_multi_var!(compact_status);
```
Comment (hzxa21) [inline src/meta/src/hummock/hummock_manager.rs:610]: @wenym1 Is it safe to not call abort here?

File: src/meta/src/hummock/error.rs (hunk 1 above)
Comment (zwang28) [reply src/meta/src/hummock/error.rs:51]: Currently `meta_store::Error::Internal` is equivalent to `EtcdError`. Need to expand `meta_store::Error::Internal` to more detail meta_store errors.

File: src/meta/src/hummock/hummock_manager.rs (hunk 2 above)
Comment (wenym1) [reply src/meta/src/hummock/hummock_manager.rs:610]: Seems that our updated rust-toolchain has no longer required us to drop variables on every branch.

**Review by hzxa21 — APPROVED** [review]

**Pull request merged**